it/flags-provider/src/test/java/com/linecorp/armeria/common/FlagsProviderTest.java

```diff
@@ -57,7 +58,6 @@ void reloadFlags() throws ClassNotFoundException {
 
     @Test
     void overrideDefaultFlagsProvider() throws Throwable {
-        assertFlags("useOpenSsl").isEqualTo(false);
```

Comment:
Could this be converted to the following?
```java
assertFlags("tlsEngineType").equals(TlsEngineType.OPENSSL);
```

Reply:
Sure!

Reply:
I guess the `TlsEngineType` was loaded from the default class loader, whereas the assertion is using the `FlagsClassLoader`.

Can you try the following?

```
user@AL02437565 armeria % git diff
diff --git a/it/flags-provider/src/test/java/com/linecorp/armeria/common/FlagsProviderTest.java b/it/flags-provider/src/test/java/com/linecorp/armeria/common/FlagsProviderTest.java
index a010976bf..619bfd3eb 100644
--- a/it/flags-provider/src/test/java/com/linecorp/armeria/common/FlagsProviderTest.java
+++ b/it/flags-provider/src/test/java/com/linecorp/armeria/common/FlagsProviderTest.java
@@ -158,11 +158,8 @@ class FlagsProviderTest {
     }
 
     private ObjectAssert<Object> assertFlags(String flagsMethod) throws Throwable {
-        final Lookup lookup = MethodHandles.publicLookup();
-        final MethodHandle method =
-                lookup.findStatic(flags, flagsMethod, MethodType.methodType(
-                        Flags.class.getMethod(flagsMethod).getReturnType()));
-        return assertThat(method.invoke());
+        Method method = flags.getDeclaredMethod(flagsMethod);
+        return assertThat(method.invoke(null));
     }
 
     private static class FlagsClassLoader extends ClassLoader {
user@AL02437565 armeria % 
```

I also realized it should actually be `isNotEqualTo` (following the previous implementation)